digitalout.py

```diff
@@ -85,47 +94,72 @@ def load_xml(self, node):
                             
                 else:
                     self.logger.warning(f"Unrecognized XML tag \'{child.tag}\' in <{self.expectedRoot}>")
+            
+            except (KeyError, ValueError):
+                self.logger.exceptions()
+                raise XMLError(self, child)
+                
                     
     def init(self):
         """
         Initialize the device hardware with the attributes set in load_xml
         """
     
-        if not (self.stop_connections or self.reset_connection):
+        if not (self.stop_connections or self.reset_connection) and self.enable:
             
-             if self.enable:
-
                 # Clear old task
                 if self.task != None:
```

Comment:
equality comparison with None

Reply:
Also self.task not set equal to None in constructor so I'm not sure if this would work either way.